src/psyclone/psyir/nodes/operation.py

```diff
@@ -180,14 +345,31 @@ def create(oper, child):
             are not of the expected type.
 
         '''
-        if not isinstance(oper, UnaryOperation.Operator):
+        if not isinstance(operator, Enum) or \
+           operator not in UnaryOperation.Operator:
             raise GenerationError(
-                f"oper argument in create method of UnaryOperation class "
+                f"operator argument in create method of UnaryOperation class "
                 f"should be a PSyIR UnaryOperation Operator but found "
-                f"'{type(oper).__name__}'.")
-
-        unary_op = UnaryOperation(oper)
-        unary_op.children = [child]
+                f"'{type(operator).__name__}'.")
+
+        unary_op = UnaryOperation(operator)
+        name = None
+        if isinstance(operand, tuple):
+            if not len(operand) == 2:
+                raise GenerationError(
+                    f"If the argument in the create method of "
+                    f"UnaryOperation class is a tuple, it's length "
+                    f"should be 2, but it is {len(operand)}.")
+            if not isinstance(operand[0], str):
+                raise GenerationError(
+                    f"If the argument in the create method of "
+                    f"UnaryOperation class is a tuple, its first "
+                    f"argument should be a str, but found "
+                    f"{type(operand[0]).__name__}.")
+            Operation._validate_name(operand[0])
```

Comment:
Again, we may not need this check if it is already done inside the call below.

Reply:
Removed for unary, binary and nary.